File: DS-Algo/inflearn_algo_basic/14/14.py
import logging
logging.basicConfig(level=logging.INFO, format='[%(levelname)s] %(message)s')

def reverse_num(n):
    '''
        reverse input n
        
        Args : 
            n(int) 
        Return :
            reversed_num(int)
    '''
    temp_n = n
    reversed_num = 0
    while temp_n > 0:
        reversed_num = temp_n % 10 + reversed_num * 10
        temp_n //= 10
    
    return reversed_num

def is_prime(n):
    ''' 
        check n is prime true of false 

        Args:
            n(int)
        Return :
            True of False
    '''
    if(n == 1) : return False
    end = int(n ** 0.5)
    for i in range(2,end+1):
        if n % i == 0:
            return False
        
    return True

N = int(input())
input_num_lst = list(map(int,input().split()))

logging.info(f'input_num_lst : {input_num_lst}')
logging.info(f'revers_num(350) : {reverse_num(350)}')
logging.info(f'is_prime(40) : {is_prime(40)}')

for value in input_num_lst:
    reversed_num = reverse_num(value)
    if is_prime(reversed_num) == True:
        print(reversed_num,end=' ')

File: DS-Algo/inflearn_algo_basic/14/14.cpp
#include <iostream>
#include <vector>

using namespace std;

/*
    @brief : reverse number  (e.g. 35 -> 53)
    @param : n(int) -> input num
    @return : reversed_num(int) -> reversed num
*/
int reverseNum(int n)
{
    int reversed_num = 0;
    int temp_n = n;
    while (temp_n)
    {
        reversed_num = reversed_num * 10 + temp_n % 10;
        temp_n = temp_n / 10;
    }

    return reversed_num;
}

/*
    @brief : Checks whether a given integer is a prime number
    @param : n(int) -> input num
    @return : true or false (bool)
*/
bool isPrime(int n)
{
    if (n == 1)
        return false;

    for (int i = 2; i * i <= n; i++)
    {
        if (n % i == 0)
            return false;
    }

    return true;
}

int main()
{
#ifdef DEBUG
    cout << "[INFO] reverseNum(350) : " << reverseNum(350) << endl;
    cout << "[INFO] isPrime(41) : " << isPrime(41) << endl;
    cout << "[INFO] isPrime(33) : " << isPrime(33) << endl;
#endif
    int N;
    cin >> N;
    vector<int> input_num_vector(N);
    int temp_input;
    for (int i = 0; i < N; i++)
    {
        cin >> temp_input;
        temp_input = reverseNum(temp_input);
        if(isPrime(temp_input) == true)
            cout << temp_input << " ";
    }

    return 0;
}